spec/presenters/user_list_presenter_spec.rb

```diff
@@ -2,54 +2,45 @@
 
 describe "UserListPresenter" do
   
-  let(:presenter) { UserListPresenter.new(type: :city) }
+  let(:presenter) { UserListPresenter.new(city: "Paris") }
   
   describe "new" do
-    context "empty location" do
-      it "sets default city" do
-        expect(UserListPresenter.new(type: :city).location).to eq("san francisco")
-      end
-      
-      it "sets default country" do
-        expect(UserListPresenter.new(type: :country).location).to eq("united states")
-      end
-      
-      it "sets default language" do
-        expect(presenter.language).to eq("JavaScript")
-      end
+    it { expect(UserListPresenter.new(city: "Paris").location).to eq("paris") }
+    it { expect(UserListPresenter.new("city" => "Paris").location).to eq("paris") }
+    it { expect(UserListPresenter.new(city: "Paris").type).to eq(:city) }
+
+    it { expect(UserListPresenter.new(country: "France").location).to eq("france") }
+    it { expect(UserListPresenter.new("country" => "France").location).to eq("france") }
+    it { expect(UserListPresenter.new(country: "France").type).to eq(:country) }
+
+    it { expect(UserListPresenter.new({}).location).to be nil }
+    it { expect(UserListPresenter.new({}).type).to eq(:world) }
+
+    it "sets default language" do
+      expect(UserListPresenter.new({}).language).to eq("JavaScript")
     end
 
     context "location provided" do
       it "location trim whitespace" do
-        expect(UserListPresenter.new(type: :city, city: " paris ").location).to eq("paris")
+        expect(UserListPresenter.new(city: " paris ").location).to eq("paris")
       end
     end
   end
   
   describe "languages" do
-    it { expect(presenter.languages.count).to eq(223) }
+    it { expect(presenter.languages.count).to eq(228) }
```

Comment:
Maybe we can put a stub here. Otherwise the tests will always fail when the file is updated. :)